numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -373,29 +374,24 @@ gentype_nonzero_number(PyObject *m1)
 static PyObject *
 gentype_str(PyObject *self)
 {
-    PyObject *arr, *ret = NULL;
-
-    arr = PyArray_FromScalar(self, NULL);
-    if (arr != NULL) {
-        ret = PyObject_Str((PyObject *)arr);
-        Py_DECREF(arr);
+    static PyObject *fallback = NULL;
+    npy_cache_import("numpy.core.arrayprint", "_scalar_fallback", &fallback);
+    if (fallback == NULL) {
+        return NULL;
     }
-    return ret;
+    return PyObject_CallFunction(fallback, "O", self);
```

Comment:
It gives
```python
>>> repr(np.datetime64('2005-02-25')[...])
"array('2005-02-25', dtype='datetime64[D]')"
```

(it used to give `"array(datetime.date(2005, 2, 25), dtype='datetime64[D]')"`)